Assets/Scripts/Game/GamePrototype.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections.Generic;
+using johnny.HexProject.Logic;
+using johnny.HexProject.Visuals;
+using UnityEngine;
+
+namespace johnny.HexProject.Game
+{
+    public class GamePrototype : MonoBehaviour
+    {
+        [SerializeField] private int _randomSeed;
+        [SerializeField] private int _numberOfPlayers;
+        [SerializeField] private GameObject _tilePrefab;
+        [SerializeField] private GameObject _playerPrefab;
+        [SerializeField] private int _boardWidth;
+        [SerializeField] private int _boardHeight;
+        
+        private bool _isGameOver;
+        private List<PlayerBehaviour> _playerBehaviours = new();
+        
+        private void Awake()
+        {
+            Random.InitState(_randomSeed);
```

Comment:
In order to guarantee that the simulation will be deterministic, it's worth it implementing a RandomGenerator class, that would also take this `_randomSeed` as seed and have functions like `Random.value` and `Random.Range`. After creating this class, it could be distributed via dependency injection or be a singleton, and then every Random call in the simulation should be replaced by it.